platform-sdk/docs/core/consensus-glossary.md

```diff
@@ -0,0 +1,199 @@
+
+The reason why the word "claimed" is used is because it's possible for a malicious node to lie about the existence
+of a parent that may not exist. The fact that a claimed parent birth round may not be correct is not a problem though.
+
+- if an attacker claims to have a parent that doesn't exist, we hold the child event in the orphan buffer until
+  the non-existent parent is ancient
+- if the attacker claims a parent has the wrong birth round and the parent exists, we detect the discrepancy and do not
+  link the child to the misconfigured parent. In this situation, it is as if the parent-child link was never created.
+
+## Branching
+
+FUTURE WORK: write down the technical definition
+
+## Consensus Generation
+
+When a round reaches consensus, all events in that round are assigned a consensus generation. For round `R`, use the
+following algorithm to assign each event a consensus generation:
+
+- iterate over events in topological order (this can be achieved by sorting on preconsensus generation)
+- for each event, assign the consensus generation as the maximum of the consensus generation of its parents plus 1
+- if an event has no parents, assign it a consensus generation of 1
+- for all parent events that did not reach consensus in round `R`, ignore them when for the purpose of this calculation
+
+Round generations are guaranteed to be deterministic across all nodes.
+
+The consensus generation is used as a tie breaker when determining the consensus order of events in a round. This
+provides the guarantee that if event A and event B both reach consensus and A is a parent of B, then A will reach
+consensus before B.
+
+After computing the consensus order for a round, the consensus generation of an event is no longer needed and is
+
+## Expired Event
+
+Analogous to the concept of an [ancient event](#ancient-event), an expired event is an event that is old enough that
+we are no longer interested in gossiping it. Not all ancient events are expired events, but all expired events are
+ancient events. It would technically be legal to consider all ancient events to be expired events, but we choose not
+to in order to effectively gossip with peers that may be slightly behind us.
+
+The exact computation for when an event becomes expired is an implementation detail of gossip algorithms. Any
+algorithm that satisfies the above definition is acceptable as long as the gossip algorithm is happy with the result.
+
+## Future Event
+
+An event is considered to be future event if its birth round is greater than the
+[pending round](#pending-consensus-round).
+
+When choosing events to use as parents for a new self event, future events are never selected. This is required to
+maintain the [invariant that birth rounds never decrease](#invariant-birth-rounds-never-decrease).
+
+### Unverifiable Future Event
+
+A future event is considered to be unverifiable future event if we do not yet know what the roster is going to be
```

Comment:
"A future event is considered to be unverifiable future event"

Should be either "is considered to be *an* unverifiable future event", or "is considered to be unverifiable"